dbuf-lsp/src/diagnostic_handler/semantic_token/token.rs

```diff
@@ -19,39 +22,11 @@ pub enum Token {
 }
 
 impl Token {
-    pub const COUNT: u32 = 10;
-
-    pub fn to_index(&self) -> u32 {
-        match &self {
-            Token::Type => 0,
-            Token::Message => 1,
-            Token::Enum => 2,
-            Token::Parameter => 3,
-            Token::Property => 4,
-            Token::EnumConstructor => 5,
-            Token::Keyword => 6,
-            Token::String => 7,
-            Token::Number => 8,
-            Token::Operator => 9,
-        }
+    pub fn to_index(self) -> u32 {
+        Token::iter().position(|t| t == self).unwrap() as u32
```

Comment:
This is an [enum discriminant](https://doc.rust-lang.org/std/mem/fn.discriminant.html#accessing-the-numeric-value-of-the-discriminant)